### PR TITLE
[BugFix] fix bunch of bugs of resource group

### LINKED ISSUE #7667: can't excute query after drop resource group
### Steps to reproduce the behavior (Required)
create resource group rg1
to
    (user='root')
with (
    'cpu_core_limit' = '4',
    'mem_limit' = '10%',
    'concurrency_limit' = '2',
    'type' = 'normal'
);
set resource_group=rg1;
drop resource group rg1;
select current_version();

### Expected behavior (Required)
`query ok`
### Real behavior (Required)
`ERROR 1064 (HY000): Invalid resource_group: rg1`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> show variables like '%resource%';
+-----------------------+-------+
| Variable_name         | Value |
+-----------------------+-------+
| enable_resource_group | true  |
| resource_group        |       |
+-----------------------+-------+
```

### LINKED ISSUE #7708: ErrorCode in fe.audit.log error
### Steps to reproduce the behavior (Required)
1.
`excute error sql`
2.
`ErrorCode=CANCELLED in fe.audit.log`
3.
`excute right sql`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 26f8840 |
+----------------------------+
```
### Expected behavior (Required)
`ErrorCode=OK`
### Real behavior (Required)
`ErrorCode=CANCELLED in fe.audit.log`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### LINKED ISSUE #7801: ResourceGroup info error in audit log when set session variable resource_group
### Steps to reproduce the behavior (Required)
```
create resource group rg1
to
    (user='root')
with (
    'cpu_core_limit' = '4',
    'mem_limit' = '10%',
    'concurrency_limit' = '10',
    'type' = 'normal'
);
set resource_group=rg1;
```
`cat fe.audit.log`
### Expected behavior (Required)
`ResourceGroup=rg1`
### Real behavior (Required)
`ResourceGroup=default_wg`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 332a886 |
+----------------------------+
```

### LINKED ISSUE #7858: can't get resource group info by api after alter resource group
### Steps to reproduce the behavior (Required)
```
create resource group rg3
to
    (user='root',db='yn_db')
with (
    'cpu_core_limit' = '4',
    'mem_limit' = '10%',
    'concurrency_limit' = '10',
    'type' = 'normal'
);
```
excute sql and get resource group info by api

```
ALTER resource group rg3 WITH (
    'big_query_cpu_second_limit' = '10',
    'big_query_scan_rows_limit' = '10000',
    'big_query_mem_limit' = '107374182'
);
```
`curl http://172.26.92.141:8538/metrics |grep starrocks_be_resource_group|grep rg3`

### Expected behavior (Required)
```
starrocks_be_resource_group_bigquery_count{name="rg3"} 0
starrocks_be_resource_group_concurrency_overflow_count{name="rg3"} 0
starrocks_be_resource_group_cpu_limit_ratio{name="rg3"} 0.026144
starrocks_be_resource_group_cpu_use_ratio{name="rg3"} 1.000000
starrocks_be_resource_group_mem_allocated_bytes{name="rg3} 0
starrocks_be_resource_group_mem_limit_bytes{name="rg3"} 4305731862
starrocks_be_resource_group_running_queries{name="rg3"} 0
starrocks_be_resource_group_total_queries{name="rg3"} 1
```
### Real behavior (Required)
```
starrocks_be_resource_group_bigquery_count{name="rg3"} 0
starrocks_be_resource_group_concurrency_overflow_count{name="rg3"} 0
starrocks_be_resource_group_cpu_limit_ratio{name="rg3"} 0.000000
starrocks_be_resource_group_cpu_use_ratio{name="rg3"} 0.000000
starrocks_be_resource_group_mem_allocated_bytes{name="rg3"} 0
starrocks_be_resource_group_mem_limit_bytes{name="rg3"} 0
starrocks_be_resource_group_running_queries{name="rg3"} 0
starrocks_be_resource_group_total_queries{name="rg3"} 0
```
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 648bbc5 |
+----------------------------+
```

### LINKED ISSUE #7860: Monitoring information for resource group is often not updated
### Steps to reproduce the behavior (Required)
```
create resource group rg4
to
    (user='root',db='yn_db')
with (
    'cpu_core_limit' = '4',
    'mem_limit' = '10%',
    'concurrency_limit' = '10',
    'type' = 'normal'
);
```
Continue to get resource group info when executing query sql
### Expected behavior (Required)
`Able to obtain  group information in time`
### Real behavior (Required)
`Monitoring information is often not updated`
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 648bbc5 |
+----------------------------+
```

### LINKED ISSUE #7862: starrocks_be_resource_group_mem_allocated_bytes in metric for resource group Display is not accurate
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
2. INSERT INTO '....'
3. SELECT  '....'

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
```
mysql> select current_version();
+----------------------------+
| current_version()          |
+----------------------------+
| BRANCH-2.3-RELEASE 648bbc5 |
+----------------------------+
```

### PR DESCRIPTION
**Pull request by murphyatwork**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #7667
Fixes #7801
Fixes #7858
Fixes #7860
Fixes #7862
Fixes #7708

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 1 / 2 (50.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|2|50.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 7 (28.57%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|5|20.00%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 9 (22.22%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|6|16.67%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 2 / 9 (22.22%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|6|16.67%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|

File: be/src/exec/workgroup/work_group.cpp [hunk 1]
```diff
@@ -351,11 +372,12 @@ void WorkGroupManager::apply(const std::vector<TWorkGroupOp>& ops) {
...
     while (it != _workgroup_expired_versions.end()) {
         auto wg_it = _workgroups.find(*it);
-        DCHECK(wg_it != _workgroups.end());
-        if (wg_it->second->is_removable()) {
+        if (wg_it != _workgroups.end() && wg_it->second->is_removable()) {
+            int128_t wg_id = *it;
             _sum_cpu_limit -= wg_it->second->cpu_limit();
             _workgroups.erase(wg_it);
             _workgroup_expired_versions.erase(it++);
+            LOG(INFO) << "cleanup expired workgroup version:  " << (int64_t)(wg_id >> 64) << "," << (int64_t)wg_id;
```
Comment (kangkaisen) [inline be/src/exec/workgroup/work_group.cpp:380]: Will print a lot of logs?

**Review by kangkaisen — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 10 (30.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|6|16.67%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

File: be/src/exec/workgroup/work_group.cpp (hunk 1 above)
Comment (mofeiatwork) [reply be/src/exec/workgroup/work_group.cpp:379]: Only log at workgroup DDL (create/alter/delete).

**Comment by mofeiatwork** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by mofeiatwork** [comment]
@Mergifyio backport branch-2.2 branch-2.3

**Review by kangkaisen — APPROVED** [review]

**Review by satanson — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 3 / 13 (23.08%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/common/Status.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/StmtExecutor.java|0|1|00.00%|
|:large_blue_circle:|com/starrocks/qe/Coordinator.java|1|8|12.50%|
|:large_blue_circle:|com/starrocks/catalog/WorkGroupMgr.java|1|2|50.00%|
|:large_blue_circle:|com/starrocks/qe/ConnectProcessor.java|1|1|100.00%|

**Pull request merged**